packages/block-editor/src/components/link-control/index.js

```diff
@@ -219,20 +253,52 @@ function LinkControl( {
 			const isInternal = startsWith( val, '#' );
 			const isTel = protocol.includes( 'tel' );
 
-			const handleManualEntry =
+			const maybeURL =
 				isInternal ||
 				isMailto ||
 				isTel ||
 				isURL( val ) ||
 				( val && val.includes( 'www.' ) );
 
-			return handleManualEntry
+			return maybeURL
 				? handleDirectEntry( val, args )
 				: handleEntitySearch( val, args );
 		},
 		[ handleDirectEntry, fetchSearchSuggestions ]
 	);
 
+	const handleOnCreate = async ( entityTitle ) => {
+		let newEntity;
+
+		setIsResolvingLink( true );
+		setErrorMsg( null );
+		try {
+			newEntity = await createEntity( 'page', entityTitle );
+		} catch ( error ) {
+			setErrorMsg(
+				error.msg ||
+					__(
+						'An unknown error occurred during Page creation. Please try again.'
```

Comment:
@aduth It feels like the `page` should not be baked into this component. Rather the consuming component (in this case Nav Link Block) should provide the "type". LinkControl should be unaware of the type of entity it is creating. Do you agree?

This was refactored by @marekhrabe so I could have things wrong.

Reply:
I would agree with your concern. Ignoring the whole "use outside WordPress" scenario, I might imagine there are even scenarios _within_ WordPress where we might want to be able to use this component and generate new.... products, real estate listings, etc.

A few things I would advise:

- We probably don't want to just pass the `'Page'` fragment and interpolate it in the string, since it can make the string harder (or impossible) to translate correctly in languages where the sentence structure could change depending on e.g. gender of the term.
- We could consider to try to phrase it in a way which removes reference to this entity type altogether (more generic "error occurred during creation").
- Worth noting as well there are many labels associated with post types that we would tend to prefer to use in UIs ([reference](https://developer.wordpress.org/reference/functions/get_post_type_labels/)), extending these as new requirements surface.
- I _think_ there are other components which accept some sort of `labels` prop for this kind of purpose, that you could reference as prior art.

Reply:
I've opted to keep it simple and remove the ref to `Page`. If we need to differentiate types later than a refactor can be handled in another PR.